app/src/main/java/com/globant/harrypotterapp/activity/CharacterDetailActivity.kt

```diff
@@ -0,0 +1,54 @@
+package com.globant.harrypotterapp.activity
+
+import android.content.Context
+import android.content.Intent
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import com.globant.domain.entity.CharacterDetail
+import com.globant.harrypotterapp.R
+import com.globant.harrypotterapp.databinding.ActivityCharacterDetailBinding
+import com.globant.harrypotterapp.viewmodel.CharacterDetailViewModel
+import org.koin.androidx.viewmodel.ext.android.viewModel
+
+class CharacterDetailActivity : AppCompatActivity() {
+
+    private lateinit var binding: ActivityCharacterDetailBinding
+    private val characterDetailViewModel by viewModel<CharacterDetailViewModel>()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivityCharacterDetailBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        characterDetailViewModel.getCharacterDetailLiveData().observe(::getLifecycle, ::updateUI)
+        characterDetailViewModel.fetchCharacterDetail()
+    }
+
+    private fun updateUI(data: CharacterDetail) {
+        with(binding) {
```

Comment:
this is not fine, call a method to update the XML info. this method should have 
when (data.state) {
     state1 -> {..}
     state2 -> {..}
}

Reply:
Done Mau, thanks!